---
slug: accessing-refget
title: Accessing Ensembl's refget services
description: How to access the refget service provided by Ensembl
related_articles:
  - href: ../../../../../about-ensembl/accessing-data/apis/refget.md
---
# Accessing Ensembl's refget service

The Ensembl refget service can be accessed at [beta.ensembl.org/data/refget/](https://beta.ensembl.org/data/refget/).

## About the standard

Refget's official documentation site can be found [here](https://github.com/samtools/hts-specs/blob/fde0b700d7d1aeb4d40cc27c820965a6f365f650/refget.md) alongside a [OpenAPI v3.0 service description](https://raw.githubusercontent.com/samtools/hts-specs/fde0b700d7d1aeb4d40cc27c820965a6f365f650/pub/refget-openapi.yaml). It provides detailed information about how to interact with implementations of the standard and in the interest of maintaining consistency, this will not be duplicated here. However, a brief example of how to access sequence data in an Ensembl specific context is provided below.

## Things to be aware of when using refget

### Coordinates used in refget

The refget protocol uses the `0-start, half-open` coordinate system, which is found in other formats such as BED, SAM/BAM/CRAM and utilised by the UCSC Genome Browser. This system is sometimes called 0-based. In this system you start counting sequence positions from 0. More information about this system is available from the [UCSC Genome Browser team's blog site](https://genome-blog.gi.ucsc.edu/blog/2016/12/12/the-ucsc-genome-browser-coordinate-counting-systems/).

When converting from Ensembl coordinates (1-start, fully-closed; sometimes called 1-based) you need only remove `1` from the start. The end remains the same. For example, given an Ensembl coordinate (1-based) 1:12714000**1**-127140001, this is represented as 1:12714000**0**-127140001 (0-based).

### Using appropriate `Accept` content-types

Refget is particular about the content-types it is provided. When requesting a sequence you must specify `plain/text` as the content type and when accessing sequence metadata you must specify `application/json`. Some URLs will fail when clicked on within a web browser due to their automated provision of Accept content types.

### Why a metadata mentions multiple aliases for Ensembl sequences rather than just one

The refget system effectively creates a non-redundant database of sequences. If the same sequence of characters appear in the same order they will result in the same checksum identifier being generated irrespective of molecule type (nucleotides versus amino acid residues).

### Use of multiple identifiers for a sequence

Refget's specification details two checksum algorithms which can be used to generate the identifier. These are a hex encoded `MD5` or a scheme called [`sha512t24u`](https://samtools.github.io/hts-specs/refget.html#refget-checksum-algorithm). A GA4GH identifier is generated by adding `ga4gh:SQ.` to a `sh512t24u` checksum e.g. `ga4gh:SQ.0qw_sn8Cl7OmMTFlukjFD2DUejW0T80Y` and `17dab79b963ccd8e7377cef59a54fe1c` represent the same sequence.

## Third party refget clients

You can use the [Python `refget`](https://pypi.org/project/refget/) library to fetch sequences from the Ensembl refget instance.

## Accessing sequence data for features in Ensembl

The ["Getting started with Graphql"](/help/articles/getting-started-with-ensembl-graph-ql-services) and ["Using Graphql"](/help/articles/using-graph-ql) pages cover how to access data through Ensembl GraphQL service and the example below assumes that this service is being used.

Before starting you must obtain a `genome_id` for the genome (assembly + annotation) of interest. We are using `"GCA_000001405.29`, which represents the human assembly GRCh38.p14 (hg38).

```text
query {
  genomes(
    by_keyword: {assembly_accession_id: "GCA_000001405.29"})
  {
    genome_id
  }
}
```

returns:

```json
{
  "data": {
    "genomes": [
      {
        "genome_id": "a7335667-93e7-11ec-a39d-005056b38ce3"
      }
    ]
  }
}
```

### Fetching the genomic sequence for a feature

Using the previously retrieved `genome_id`, we can search for a feature (such as a gene) you must find the checksum for a feature's (gene, transcript etc.) region, and its start and end coordinates via `slice`

```text
query {
  genes(
    by_symbol: {genome_id: "a7335667-93e7-11ec-a39d-005056b38ce3", symbol: "JAG1"}
  ) {
    slice{
      region{
        sequence{
          checksum
        }
      }
      location{
        start
        end
      }
    }
  
  }
}
```

returns:

```json
{
  "data": {
    "genes": [
      {
        "slice": {
          "region": {
            "sequence": {
              "checksum": "b18e6c531b0bd70e949a7fc20859cb01"
            }
          },
          "location": {
            "start": 10637684,
            "end": 10673999
          }
        }
      }
    ]
  }
}
```

Then use the information from the previous step to query refget (Curl for clarity). Note you must convert from Ensembl coordinates (1-based) to 0-based coordinates.

```bash
curl -H'Accept: text/plain' "https://beta.ensembl.org/data/refget/sequence/b18e6c531b0bd70e949a7fc20859cb01?start=10637683&end=10673999"

```

returns:

```text
GGCCATTAATCCAGTGGTGTTTATT ... CCCGGAGAGCCCGTCT 
```

### Fetching cDNA, CDS and protein sequences

We can use the following GraphQL query to retrieve all known transcripts for the human JAG1 gene using the previously retrieved `genome_id`.

```text
query JAG1 {
  genes(
    by_symbol: {
      symbol: "JAG1",
      genome_id: "a7335667-93e7-11ec-a39d-005056b38ce3"
    }
  ) {
    transcripts {
      stable_id
      symbol
      metadata {
        canonical {
          value
        }
      }
      product_generating_contexts {
        product_type
        cdna {
          sequence {
            checksum
          }
        }
        cds {
          sequence {
            checksum
          }
        }
        product {
          stable_id
          sequence {
            checksum
          }
        }
      }
    }
  }
}
```

Each sequence product (cDNA, CDS and protein) refer to a checksum. The value of this field should be used verbatim with the refget service.

```json
{
  "data": {
    "genes": [
      {
        "transcripts": [
          {
            "stable_id": "ENST00000254958.10",
            "symbol": "JAG1-201",
            "metadata": {
              "canonical": {
                "value": true
              }
            },
            "product_generating_contexts": [
              {
                "product_type": "Protein",
                "cdna": {
                  "sequence": {
                    "checksum": "d8d95cac5218b333fba43ed6ebe4b017"
                  }
                },
                "cds": {
                  "sequence": {
                    "checksum": "27ec2b034fb386c69f48653d2f18daf2"
                  }
                },
                "product": {
                  "stable_id": "ENSP00000254958.4",
                  "sequence": {
                    "checksum": "84706f0ee2a4d23ae4050036e62dac5d"
                  }
                }
              }
            ]
          },
          ...
      },
      ...
    ]
  }
}
```

Each sequence can now be retrieved with the following URLs.

- cDNA: [beta.ensembl.org/data/refget/sequence/d8d95cac5218b333fba43ed6ebe4b017](https://beta.ensembl.org/data/refget/sequence/2f8c687a656debd4c8b1b4d47747ba35)
- CDS: [beta.ensembl.org/data/refget/sequence/27ec2b034fb386c69f48653d2f18daf2](https://beta.ensembl.org/data/refget/sequence/c6b5e532b5aef696e987ca55a538ae9e)
- Protein: [beta.ensembl.org/data/refget/sequence/84706f0ee2a4d23ae4050036e62dac5d](https://beta.ensembl.org/data/refget/sequence/92addb948c6c652abc1dcecca05f26c0)

Metadata concerning each of the records can be retreived by adding a `/metadata` to each URL.

- cDNA: [beta.ensembl.org/data/refget/sequence/d8d95cac5218b333fba43ed6ebe4b017/metadata](https://beta.ensembl.org/data/refget/sequence/2f8c687a656debd4c8b1b4d47747ba35/metadata)
- CDS: [beta.ensembl.org/data/refget/sequence/27ec2b034fb386c69f48653d2f18daf2/metadata](https://beta.ensembl.org/data/refget/sequence/c6b5e532b5aef696e987ca55a538ae9e/metadata)
- Protein: [beta.ensembl.org/data/refget/sequence/84706f0ee2a4d23ae4050036e62dac5d/metadata](https://beta.ensembl.org/data/refget/sequence/92addb948c6c652abc1dcecca05f26c0/metadata)

```bash
# Get the sequence metadata
curl -H'Accept: application/json' 'https://beta.ensembl.org/data/refget/sequence/84706f0ee2a4d23ae4050036e62dac5d/metadata'
{"metadata": {"aliases": [
  {"alias": "ga4gh:SQ.0pyg0Uu3J8LDWTEm_14sUQ7DCX8rK8nl", "naming_authority": "ga4gh"},
  {"alias": "ENSP00000254958.4", "naming_authority": "Ensembl"},
  {"alias": "ENSP04835022723.1", "naming_authority": "Ensembl"}, ... ],
"length": 1218, "md5": "84706f0ee2a4d23ae4050036e62dac5d", "trunc512": "d29ca0d14bb727c2c3593126ff5e2c510ec3097f2b2bc9e5"}}

# Pipe into wc and count characters to confirm length
curl -s -H'Accept: text/plain' 'https://beta.ensembl.org/data/refget/sequence/84706f0ee2a4d23ae4050036e62dac5d' | wc -c
    1218

# Retreiving the first 10 residues of the sequence
curl -s -H'Accept: text/plain' 'https://beta.ensembl.org/data/refget/sequence/84706f0ee2a4d23ae4050036e62dac5d?start=0&end=10'
MRSPRTRGRS
```
