---
slug: transcripts
title: What’s in the Transcripts view?
description: How to browse and download transcript data
related_articles:
  - href: entity-viewer.md
  - href: gene-function.md
  - href: overview.md
  - href: external-references.md
tags:
  - entity
status: draft
---

# What’s in the Transcripts view?

Transcript view lists all transcripts of the focus gene. You can find the graphical deception of the transcript models with exons illustrated as boxes connected by lines representing introns. Coding exons are coloured, while non-coding exons are empty.
 
Transcript models are positioned relative to the gene start and end, so that the yellow grid matches the coordinates in the gene image above. Click on the transcript image or its corresponding Ensembl transcript stable IDs on the right to show/hide information about this transcript. The first transcript on the list is expanded by default.
 
The expanded view includes transcript name, genomic coordinates, biotype, exon count, combined exon length in bp and amino acid length of its translation for coding transcripts. Ensembl protein stable ID of the corresponding translation links out to the Gene function view. Transcript flags, such as Ensembl canonical or APPRIS can be found on the left. There is also an option to view this transcript in the Genome browser. 
 
Expanded transcript view allows quick sequence Download for both the gene and the focus transcript. Click Download to reveal customisation options, select your preference by ticking the boxes and hit the green Download button to export sequence data in FASTA format. You can select multiple options at once.
 
You can manipulate the transcript list by clicking Filter & sort on the left. The expanded menu offers options to filter the list by transcript biotype (multiple choice) and to sort by exon length and count. Changes to the transcript list in this tab will be applied to all subsequent views.
