---
slug: responsible-handover
title: Responsible Handover Framework
description: Responsible handover of Ensembl’s data, tools and technologies to users.
---

# Responsible handover of Ensembl’s data, tools and technologies to users.

The Ensembl project is committed to supporting a facilitated responsible handover of data, tools and technologies to users.
We are basing  our responsible handover  of data, tools and  technologies  on the principles of  The Responsible Handover Framework for  Artificial Intelligence  developed by [Sense about Science](https://senseaboutscience.org/).

Sense About Science's Responsible Handover Framework can be found [here](https://senseaboutscience.org/responsible-handover-of-ai/).

Questions from the framework that are relevant to the Ensembl project are addressed below to support a facilitated responsible handover.

The question references ('RH X.Y') correspond to the references in the [implementation guide](https://senseaboutscience.org/wp-content/uploads/2024/11/Responsible-Handover-implementation-guide.pdf).


## User (adopter)
To responsibly use the Ensembl tool we recommend all users clearly read the Responsible Handover Framework that we have filled in as a 'tool proposer', which can be found below.

We also recommend that when users use data from Ensembl in their research they clearly consider and record:
1. What you are using the tool for, and ensuring it lines up with what we have stated it can do (RH 1.1);
2. Clearly stating the limitations of the tool with respect to the research you're carrying out (RH 1.1.2);
3. Ensure you have reviewed training materials and resources to use the tool (RH 3.5-3.6).
4. Ensure you have understood the procedure for publishing work connected to the tool (RH 5.2.2)


## EBI (proposer)

### Tier 1: What is the purpose of the tool?
The Ensembl project has three goals: 
1. to provide a comprehensive source of stable genome annotation
2. to enable genomic interpretation
3. to support researcher driven analysis (RH 1.1). 

Ensembl generates and aggregates genome annotation such as genes, genomic variants, regulatory elements and comparative genomics for vertebrates and non-vertebrates, and makes this information freely available online (RH 1.1.1). 

The data in Ensembl are created and provided in order to support research in human, animal and plant health, as well as biodiversity (RH 1.1.2).


### Risk Assessment A
Ensembl aims to provide stable, accurate, interoperable data spanning multiple taxa to support research across the tree of life. This is executed through integrated releases and an archiving policy ensuring long term access (RH A1).

All genome sequences  released in Ensembl are subject to the [browser agreement](https://beta.ensembl.org/about/articles/browser-genome-release-agreement) and as such must be registered in INSDC. This is inline with other genome browsers such as [University of Santa Cruz (UCSC)](https://genome.ucsc.edu/) and [National Centre for Biotechnology Information (NCBI)](https://www.ncbi.nlm.nih.gov/datasets/genome/). It is worth noting that Ensembl is differentiated by its integrated release and archiving [policy](https://www.ensembl.org/info/website/archives/index.html). Please note that archives are not currently available on the [new Ensembl site](https://beta.ensembl.org/), but will be in the future. (RH A1.1)

Ensembl imposes no restrictions on access to, or use of, the data generated and the software used to analyse and present it. Ensembl data generated by members of the project are available without restriction. Ensembl code written by members of the project is provided under the Apache 2.0 licence. Training materials produced by Ensembl Outreach are provided under a CC-BY 4.0 licence.

Some of the data and software included in the distribution may be subject to third-party constraints. Users of the data and software are solely responsible for establishing the nature of and complying with any such restrictions.

The [European Molecular Biology Laboratory's European Bioinformatics Institute (EMBL-EBI)](https://www.ebi.ac.uk/) provides this data and software in good faith, but make no warranty, express or implied, nor assume any legal liability or responsibility for any purpose for which they are used (RH A1.2).

Ensembl fully supports FAIR principles and integrates them into its data as far as possible. Additional information about this can be found at [fairsharing.org](https://fairsharing.org/1867) (RH A3).


### Tier 2: Where has the data come from?
Publicly available data is used to develop the automated gene annotation processes used by Ensembl. More information about gene annotation can be found [here](https://beta.ensembl.org/help/articles/gene-annotation) (RH 2.1).

Inline with the [Genome Browser Agreement](https://beta.ensembl.org/about/articles/browser-genome-release-agreement) all genome assemblies  available in Ensembl must be registered with [INSDC](https://www.insdc.org/) so the base assembly data is accessible via the data archives (such as [The European Nucleotide Archive (ENA)](https://www.ebi.ac.uk/ena/browser/home)) (RH 2.1.1).

Any bias present in the data available in Ensembl is symptomatic of the biases in the data sets ingested. Where possible efforts are made to provide data representative of the populations it describes (RH 2.2). The project aims to provide data for a broad array of taxa with key species (e.g. Human, Wheat, Sheep) having additional depth to support the needs of the research communities we support (RH 2.3).

Ensembl gene annotation pipelines are routinely validated through benchmarking against high-quality manually curated reference sets (e.g. [GENCODE](https://www.gencodegenes.org/) for human and mouse), inter-database comparisons, and community feedback. Performance metrics, such as sensitivity and specificity in gene prediction, are monitored across multiple species. Annotations produced by others (e.g. community-contributed gene models or [RefSeq](https://www.ncbi.nlm.nih.gov/refseq/) datasets) may also be integrated after undergoing quality checks to ensure compatibility and avoid redundancy. Where discrepancies exist between annotations, these are highlighted and curated where appropriate (RH 2.2.1).

Data gaps can exist due to assembly limitations, sequencing depth, or lack of transcriptomic support for specific regions or species. These are most pronounced in non-model organisms and less well-characterised genomes. Confidence scores and annotation flags (e.g. low-confidence, partial, or pseudogene) are used to indicate uncertainty in gene annotations. Where local gaps are known, they are assessed to determine their impact on downstream analyses. For key species, targeted curation or community annotation is used to resolve critical issues. Overall, the pipelines are robust to such gaps, and annotation performance remains reliable for most intended applications (RH 2.3.1).


### Tier 3: What assumptions are being made?
The gene annotation pipelines assume that core biological principles (e.g. canonical splice site patterns, coding potential, orthology relationships) are generally conserved within and across species. Transcript models are built using rules derived from well-characterised species (e.g. human, mouse), and these assumptions are transferred to less well-annotated genomes. The pipelines also assume the accuracy of the input genome assembly and supporting evidence (e.g. RNA-Seq), which may introduce limitations when these inputs are incomplete or erroneous (RH 3.1).

These pipelines incorporate known relationships such as conserved protein domains, homology, synteny, splice junction patterns, and known gene structures from reference species. However, some biological features, such as non-canonical splicing, RNA editing, or lineage-specific gene innovation, may not be fully captured (RH 3.1.1).

With factors like sample-specific expression, rare transcript variants, low-abundance genes and complex structural variation may not be fully resolved, particularly in under-characterised species or assemblies with limited evidence. These omissions can affect the completeness or accuracy of gene models in specific contexts, such as functional genomics studies or evolutionary analyses (RH 3.1.2).

The automated nature of the Ensembl gene annotation process means that annotations are only as good as the underlying genome assembly and available evidence. Known limitations include incomplete transcript coverage, assembly artefacts, misassembled regions, and biases toward well-characterised genes. Each gene model is annotated with confidence levels, biotypes, and flags indicating issues like partial coverage or uncertain structure, to help users interpret the results with appropriate caution. These limitations are regularly communicated in release notes and documentation (RH 3.3).


### Tier 4: Is the tool good enough for this use?
The performance of the Ensembl site and its supporting services is monitored internally with a KPI of 99% uptime (RH 4.1-4.2). Error rates (for the Ensembl site) are also monitored (RH 4.2.1). Known bugs are published on the [Ensembl blog known bugs page](https://www.ensembl.info/known-bugs/) (RH 4.4). If any issues are encountered with access, data or performance, help desk tickets can be raised via the "Contact us" button on the website (RH 4.1.1). Where possible, the Ensembl team will resolve issues raised (RH 4.3). The availability of the current Ensembl site can be viewed [here](https://status.uptrends.com/d29e02c5ea8a4da4a4091ab77460f443), with the new Ensembl site being added in the future (RH 4.2.2).

In the event of issues being detected with data published on the Ensembl site, a known bug will be submitted and the data will be fixed in a subsequent release or removed if it is not part of an integrated release (RH 4.4).

A new version of the Ensembl site will be released as new features are developed and tested, with significant updates being detailed on the [Ensembl blog](https://www.ensembl.info/). New data are released to the new Ensembl site approximately every two weeks. Periodically, an integrated release is produced which forms a stable set of interoperable data which will not change (RH 4.5)


### Risk Assessment B
To ensure long term data stability, only data linked to integrated releases should be used in research and analysis. Partial releases are subject to change, with the potential of data being made inaccessible in the future (if more complete data are released). (RH B1)


### Tier 5: Transparency, licenses, availability & documentation
All code for the Ensembl website is available on [GitHub](https://github.com/ensembl/) (RH 5.1). Our code is typically released under Apache 2.0, but please ensure you check the license before using it (RH 5.2, RH 5.2.1). Further information about licensing and usage can be found [here](https://beta.ensembl.org/about/articles/software-license) (RH 5.3). The code used for gene and repeat annotation and analysis can be found [here](https://github.com/Ensembl/ensembl-analysis).

Information about how to use Ensembl can be found in our [help pages](https://beta.ensembl.org/help) and information about the project is available [here](https://beta.ensembl.org/about) (RH 5.4). Any additional questions can be sent to the Ensembl helpdesk using the "contact us" link on the website (RH 5.5). There are also numerous workshops run by the Ensembl Outreach team, who also produce training materials for the site. Information about this can be accessed via the [training pages](https://training.ensembl.org/) (RH 5.6).
